django/db/migrations/serializer.py

```diff
@@ -246,11 +247,13 @@ class RegexSerializer(BaseSerializer):
     def serialize(self):
         imports = {"import re"}
         regex_pattern, pattern_imports = serializer_factory(self.value.pattern).serialize()
-        regex_flags, flag_imports = serializer_factory(self.value.flags).serialize()
+        # re.UNICODE is implicit flag, hence we need to turn it off
+        flags = self.value.flags ^ re.UNICODE
```

Comment:
We can use `re.compile('').flags` instead of `re.UNICODE` but xor is still needed. Flags is `int` value e.g. `re.UNICODE -> 32, re.S -> 16, re.S | re.UNICODE -> 48`, so if we get `self.value.flags==48` we need to return `16`.

It seems that regexes with the same implicit and explicit flags are not the same:
```python
>>> re.compile(r'^\w+$', flags=re.UNICODE)==re.compile(r'^\w+$')
False
>>> re.compile(r'^\w+$', flags=re.UNICODE).flags==re.compile(r'^\w+$').flags
True
```